# Wrangling Raw Mitochondrial Genomic Gata

ğğğ±ğ­ğ’ğğª ğŸğŸğŸğŸ is a cutting-edge sequencing platform from Illumina that provides high-throughput sequencing with exceptional accuracy and speed. The platform often employs a shotgun sequencing approach, which randomly breaks the DNA into small fragments that are then sequenced. This approach is particularly effective for comprehensive genome analysis and metagenomic studies.

Our lab received data from the ğğğ±ğ­ğ’ğğª ğŸğŸğŸğŸ that was already demultiplexed, but the samples were merged due to a prior issue. This can be quite a challenge, but I solved it with a custom script. Hereâ€™s a brief overview of the process:

ğ”ğ§ğœğ¨ğ¦ğ©ğ«ğğ¬ğ¬ ğ…ğ¢ğ¥ğğ¬: The raw data files are in .gz format. The first step is to uncompress these files to access the .fastq files inside.

ğŒğšğ©ğ©ğ¢ğ§ğ  ğ…ğ¢ğ¥ğ ğˆğ§ğ­ğğ ğ«ğšğ­ğ¢ğ¨ğ§: A mapping file containing the sample names and their corresponding indexes is used. This is essential for identifying and separating the merged samples.

ğ’ğğšğ«ğœğ¡ ğ©ğğ« ğ‡ğğšğğğ«: The script searches through each .fastq read header in the uncompressed files, using the indexes from the mapping file to correctly identify and extract the individual samples.

ğ†ğğ§ğğ«ğšğ­ğ ğ’ğğ©ğšğ«ğšğ­ğ ğ…ğ¢ğ¥ğğ¬: Finally, the script outputs separate .fastq reads for the corresponding samples, ensuring that the data is correctly organized and ready for downstream analysis.

This solution effectively solves the problem in an easy way, ensuring that the sequencing data is correctly organized and ready for analysis.
